packages/hebao_v1/contracts/modules/security/GuardianUtils.sol

```diff
@@ -0,0 +1,165 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.5.11;
+
+import "./SecurityModule.sol";
+
+
+/// @title GuardianType
+/// @author Brecht Devos - <brecht@loopring.org>
+library GuardianType
```

Comment:
[Solidity doesn't allow assigning values to enums unfortunately](https://github.com/ethereum/solidity/issues/4638). The alternative would be to use an enum and do the bit shift every time or create a function for it and maybe wrap it again in new functions, but I wouldn't really call that an improvement compared to how it's done now really. :)

Reply:
I see. Leave it as-is then.

Reply:
if we don't allow a guardian address to have more than one type, then we can use "==" to compare instead of "& TYPE != 0", then we don't need the types to have the specific shifted values then.

Reply:
I would allow more types, so things like `SelfControlled` + `Hardware` are possible. But a guardian can only be put in the list once. This way a guardian has only a single `types` value (with potentially multiple bits set) which allows for easy filtering.

Reply:
ok